File: external/boost_1.55.0/libs/python/class.cpp
// Distributed under the Boost Software License, Version 1.0. (See
// accompanying file LICENSE_1_0.txt or copy at
// http://www.boost.org/LICENSE_1_0.txt)
#include <boost/python/module.hpp>
#include <boost/python/def.hpp>
#include <boost/python/object.hpp>
#include <boost/python/class.hpp>

using namespace boost::python;

struct X
{
    int x;
    X(int n) : x(n) { }
};

int x_function(X& x)
{   return x.x;
}


BOOST_PYTHON_MODULE(class_ext)
{
    class_<X>("X", init<int>());
    def("x_function", x_function);
}

#include "module_tail.cpp"

File: external/boost_1.55.0/libs/python/test/class.py
# Distributed under the Boost
# Software License, Version 1.0. (See accompanying
# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
'''
>>> from class_ext import *

Ensure sanity:

    >>> x = X(42)
    >>> x_function(x)
    42

Demonstrate extraction in the presence of metaclass changes:

    >>> class MetaX(X.__class__):
    ...     def __new__(cls, *args):
    ...         return super(MetaX, cls).__new__(cls, *args)
    >>> class XPlusMetatype(X):
    ...     __metaclass__ = MetaX
    >>> x = XPlusMetatype(42)
    >>> x_function(x)
    42


'''

def run(args = None):
    import sys
    import doctest

    if args is not None:
        sys.argv = args
    return doctest.testmod(sys.modules.get(__name__))

if __name__ == '__main__':
    print "running..."
    import sys
    status = run()[0]
    if (status == 0): print "Done."
    sys.exit(status)
